### DISCUSSION TITLE
problem v-if v-show on preview and generate

### DISCUSSION THREAD

**Discussion by zupcode** [nuxt/nuxt#19226] [Questions]
Hello everybody !

I've finish my project, everything work fine.
But when I made npm run preview the tags v-if and v-show doesn't work as on a simple mpn run dev.
I tried a npm run generate and put it online but it's the same... the div and components with v-if or v-show doesn't appear.

Do I have to configure somthig on nux.config ?

If someone have an idea...?
Thanks

**Comment by mctweb** [comment]
Can you provide a reproduction to see your code?

**Comment by zupcode** [comment]
I managed to locate the problem but I don't have the solution yet...
It comes from a @click that calls a function.
I've something like that :

It works on both :
```
<template>
  <div id="test">
    <button @click="showHide=!showHide">show/hide</button>
    <br>
    <br>
    <div v-show="showHide">Hello</div>
  </div>
</template>

<script setup>
const showHide = useShowHide()
</script>

```


It works on run dev but doesn't work on generate :
```
<template>
  <div id="test">
    <button @click="chg()">show/hide</button>
    <br>
    <br>
    <div v-show="showHide">Hello</div>
  </div>
</template>

<script setup>
const showHide = useShowHide()
function chg() {
  this.showHide = !this.showHide
}
</script>
```


But I would like to call a function to add a delay like :
```
function chg() {
  this.showHide = true
  setTimeout(() => this.showHide = false, 1000)
}
```

**Comment by mctweb** [comment]
Can we also see the useShowHide composable?
But as a start, change 'this.showHide' to 'showHide'.

**Comment by zupcode** [comment]
If I change 'this.showHide' to 'showHide' in the function, it doesn't work at all.
In the template ok but not in the script function...

Here is the composable useShowHide (in states.ts) :
`export const useShowHide = () => useState('showHide', () => false)`

**Comment by mctweb** [comment]
Composable 

```
  const showHide = useState('showHide', ()=> false)

  export const useShowHide = () => {
  return showHide
  }
```

Component (showHide is a ref, so you need to use .value )
```
<template>
  <div id="test">
    <button @click="chg()">show/hide</button>
    <br>
    <br>
    <div v-show="showHide">Hello</div>
  </div>
</template>


<script setup>
const showHide = useShowHide()
function chg() {
  showHide.value = !showHide.value
}
</script>
```

However, you can just do this ( useState is already a reusable composable function ) ( You don't need .value in the template ):
Component
```
<template>
  <div id="test">
    <button @click="showHide = !showHide">show/hide</button>
    <br>
    <br>
    <div v-show="showHide">Hello</div>
  </div>
</template>

<script setup>
const showHide = useState(‘showHide’,()=>false)
</script>
```

**Comment by zupcode** [comment]
Thank you mctweb,
it works !!!
